x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/MlAnomaliesIndexUpdate.java

```diff
@@ -232,4 +253,40 @@ static boolean isAnomaliesReadAlias(String aliasName) {
         // which is not a valid job id.
         return MlStrings.isValidId(jobIdPart);
     }
+
+    /**
+     * Strip any suffix from the index name and find any other indices
+     * that match the base name. Then return the latest index from the
+     * matching ones.
```

Comment:
I think this doesn't have to be in the Javadoc. The name explains well enough what behavior is expected from the function. You can put this comment inside the function implementation.